generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt

```diff
@@ -10,84 +10,45 @@ import kotlin.random.Random
 /**
  * Returns a generator of String of length between [minLength] and [maxLength] (inclusive)
  *
- * @param charset Set of character to be used in generated strings
- * @param exclude Characters to exclude from generated strings
+ * @param charGenerator A generator of characters which will be used to construct the strings
  */
 fun Generator.Companion.strings(
     minLength: Int = 0,
     maxLength: Int = maxOf(minLength, KWIK_DEFAULT_MAX_SIZE),
-    charset: Set<Char> = StringCharSets.printable,
-    exclude: Set<Char> = emptySet()
+    charGenerator: Generator<Char> = Generator.characters(),
 ): Generator<String> {
     require(minLength >= 0) { "Invalid minLength: $minLength" }
     require(maxLength >= minLength) { "Invalid maxLength: $minLength (minLength=$minLength)" }
 
-    val characters = (charset - exclude).toList()
-
     val generator = Generator { rng: Random ->
-        String(CharArray(rng.nextInt(minLength, maxLength + 1)) { characters.random(rng) })
+        CharArray(rng.nextInt(minLength, maxLength + 1)) { charGenerator.generate(rng) }.concatToString()
     }
 
     val samples = mutableListOf<String>()
 
     if (minLength == 0)
         samples += ""
 
-    if (' ' in characters && minLength <= 1 && maxLength >= 1)
-        samples += " "
-
     return generator.withSamples(samples)
 }
 
 /**
  * Returns a generator of String of length between 1 and [maxLength] (inclusive)
  *
- * @param charset Set of character to be used in generated strings
- * @param exclude Characters to exclude from generated strings
+ * @param charGenerator A generator of characters which will be used to construct the strings
  */
 fun Generator.Companion.nonEmptyStrings(
     maxLength: Int = KWIK_DEFAULT_MAX_SIZE,
-    charset: Set<Char> = StringCharSets.printable,
-    exclude: Set<Char> = emptySet()
-): Generator<String> = strings(1, maxLength, charset, exclude)
+    charGenerator: Generator<Char> = Generator.characters(),
+): Generator<String> = strings(1, maxLength, charGenerator)
 
 /**
  * Returns a generator of non-blank String of length between 1 and [maxLength] (inclusive)
  *
- * @param charset Set of character to be used in generated strings
- * @param exclude Characters to exclude from generated strings
+ * @param charGenerator A generator of characters which will be used to construct the strings
  */
 fun Generator.Companion.nonBlankStrings(
     maxLength: Int = KWIK_DEFAULT_MAX_SIZE,
-    charset: Set<Char> = StringCharSets.printable,
-    exclude: Set<Char> = emptySet()
-): Generator<String> = nonEmptyStrings(maxLength, charset, exclude).filterNot { it.isBlank() }
-
-/**
- * Common set of character to be used with string generators
- */
-object StringCharSets {
```

Comment:
Please leave it, and only add a deprecation annotation (`@Deprecated`).

Reply:
Restored it, along with its' tests